rootex/core/resource_loader.cpp

```diff
@@ -91,6 +91,102 @@ int ResourceLoader::Preload(ResourceCollection paths, Atomic<int>& progress)
 	Application::GetSingleton()->getThreadPool().submit(tasks);
 
 	return tasks.size() - 1; // One less for the dummy task
+}
+
+int ResourceLoader::Export(const String& sceneName, const String& sceneFilePath, Atomic<int>& progress)
+{
+	Vector<Pair<String, String>> toCopy = {
+		{ "build/game/Release/Game.exe",
+		    "Game.exe" },
+		{ "build/game/Release/alut.dll",
+		    "alut.dll" },
+		{ "build/game/Release/OpenAL32.dll",
+		    "OpenAL32.dll" },
+		{ "build/game/Release/wrap_oal.dll",
+		    "wrap_oal.dll" },
+		{ "rootex.root", "rootex.root" },
+		{ "rootex/vendor/Debugger/Debugger.lua", "rootex/vendor/Debugger/Debugger.lua" },
+		{ "game/startup.lua", "game/startup.lua" },
+		{ "rootex/vendor/ASSAO/ASSAO.hlsl", "rootex/vendor/ASSAO/ASSAO.hlsl" },
+		{ "rootex/vendor/Middleclass/Middleclass.lua", "rootex/vendor/Middleclass/Middleclass.lua" },
+		{ "THIRDPARTY.md", "THIRDPARTY.md" }
+	};
+
+	String currExportDir;
+	int i = 0;
+	do
+	{
+		currExportDir = "exports/" + sceneName;
+		if (i != 0)
+		{
+			currExportDir += std::to_string(i);
+		}
+		currExportDir += "/";
+		i++;
+	} while (OS::IsExists(currExportDir));
+
+	OS::CreateDirectoryName(currExportDir);
+	OS::CreateDirectoryName(currExportDir + "game/assets/");
+	OS::CreateDirectoryName(currExportDir + "rootex/assets/");
+
+	JSON::json gameConfig = JSON::json::parse(ResourceLoader::CreateTextResourceFile("game/game.app.json")->getString());
+	gameConfig["startLevel"] = sceneFilePath;
+	Ref<TextResourceFile> newGameConfig = ResourceLoader::CreateNewTextResourceFile(currExportDir + "game/game.app.json");
+	newGameConfig->putString(gameConfig.dump(4));
+
+	if (!newGameConfig->save())
+	{
+		WARN("Could not save application settings file");
+		OS::DeleteDirectory(currExportDir);
+		return false;
+	}
+
+	Ref<TextResourceFile> readme = ResourceLoader::CreateNewTextResourceFile(currExportDir + "readme.txt");
```

Comment:
`README.md`?

Reply:
that contains stuff about the engine, won't make sense to ship with the game.
also a lot of game users won't even know how to open an MD file